a10_octavia/controller/worker/tasks/a10_database_tasks.py

```diff
@@ -690,13 +639,42 @@ def execute(self, member_list):
             for member in member_list:
                 if member.subnet_id not in member_subnet:
                     pool_count_subnet = self.member_repo.get_pool_count_subnet(
-                        db_apis.get_session(), member.project_id, member.subnet_id)
+                        db_apis.get_session(), partition_project_list, member.subnet_id)
                     lb_count_subnet = self.loadbalancer_repo.get_lb_count_by_subnet(
-                        db_apis.get_session(), member.project_id, member.subnet_id)
+                        db_apis.get_session(), partition_project_list, member.subnet_id)
                     if pool_count_subnet <= 1 and lb_count_subnet == 0:
                         subnet_list.append(member.subnet_id)
                     member_subnet.append(member.subnet_id)
             return subnet_list
         except Exception as e:
             LOG.exception("Failed to get subnet list for members: %s", str(e))
             raise e
+
+
+class GetProjectsForPartition(BaseDatabaseTask):
+    """
+    On the basis of whether multi-tenancy settings are enabled with use parent partition,
+    the list of projects using partition will be returned.
+    """
+    def execute(self, lb_resource):
+        try:
+            l2_project = []
+            project_id = lb_resource.project_id
+            partition_name = self.vthunder_repo.get_partition_for_project(
+                db_apis.get_session(), project_id=project_id)
+            if not partition_name:
+                # There is a case when error occured while creating LB causes vthunder
+                # entry to be removed, hence need to fetch partition name using parent project id
+                parent_project_id = utils.get_parent_project(project_id)
+                if parent_project_id:
+                    partition_name = parent_project_id[:14]
+                l2_project.append(project_id)
```

Comment:
As the `partition_project_list` can be got using `get_project_list_using_partition` method which requires only the current partition_name to be known, for getting all project ids under that partition, why again this `l2_project[]` is required?

Reply:
I observed a case when vthunder entry getting deleted via revert when the creation of lb fails. In that case, I explicitly insert the project in operation in the project's list.

Reply:
Added changes.